x/incentive/types/msg_test.go

```diff
@@ -1,72 +1,523 @@
-			multiplierName: "medium",
-			expectPass:     true,
+			name: "medium multiplier is valid",
+			msgArgs: msgArgs{
+				sender:         validAddress,
+				receiver:       validAddress,
+				multiplierName: "medium",
+			},
+			expect: expectedErr{
+				pass: true,
+			},
 		},
 		{
-			from:           sdk.AccAddress(crypto.AddressHash([]byte("KavaTest1"))),
-			multiplierName: "small",
-			expectPass:     true,
+			name: "small multiplier is valid",
+			msgArgs: msgArgs{
+				sender:         validAddress,
+				receiver:       validAddress,
+				multiplierName: "small",
+			},
+			expect: expectedErr{
+				pass: true,
+			},
 		},
 		{
-			from:           sdk.AccAddress{},
-			multiplierName: "medium",
-			expectPass:     false,
+			name: "invalid sender",
+			msgArgs: msgArgs{
+				sender:         sdk.AccAddress{},
+				receiver:       validAddress,
+				multiplierName: "medium",
+			},
+			expect: expectedErr{
+				wraps: sdkerrors.ErrInvalidAddress,
+			},
 		},
 		{
-			from:           sdk.AccAddress(crypto.AddressHash([]byte("KavaTest1"))),
-			multiplierName: "huge",
-			expectPass:     false,
+			name: "invalid receiver",
+			msgArgs: msgArgs{
+				sender:         validAddress,
+				receiver:       sdk.AccAddress{},
+				multiplierName: "medium",
+			},
+			expect: expectedErr{
+				wraps: sdkerrors.ErrInvalidAddress,
+			},
 		},
+		{
+			name: "invalid multiplier",
+			msgArgs: msgArgs{
+				sender:         validAddress,
+				receiver:       validAddress,
+				multiplierName: "huge",
+			},
+			expect: expectedErr{
+				wraps: types.ErrInvalidMultiplier,
+			},
+		},
+		{
+			name: "multiplier with capitalization is invalid",
+			msgArgs: msgArgs{
+				sender:         validAddress,
+				receiver:       validAddress,
+				multiplierName: "Large",
+			},
+			expect: expectedErr{
+				wraps: types.ErrInvalidMultiplier,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+
+			msg := types.NewMsgClaimUSDXMintingRewardVVesting(tc.msgArgs.sender, tc.msgArgs.receiver, tc.msgArgs.multiplierName)
+
+			err := msg.ValidateBasic()
+			if tc.expect.pass {
+				require.NoError(t, err)
+			} else {
+				require.Truef(t, errors.Is(err, tc.expect.wraps), "expected error '%s' was not actual '%s'", tc.expect.wraps, err)
+			}
+		})
 	}
-	suite.tests = tests
 }
 
-func (suite *MsgTestSuite) TestMsgValidation() {
-	for _, t := range suite.tests {
-		msg := types.NewMsgClaimUSDXMintingReward(t.from, t.multiplierName)
-		err := msg.ValidateBasic()
-		if t.expectPass {
-			suite.Require().NoError(err)
-		} else {
-			suite.Require().Error(err)
-		}
+func TestMsgClaimUSDXMintingReward_Validate(t *testing.T) {
+	validAddress := sdk.AccAddress(crypto.AddressHash([]byte("KavaTest1")))
+
+	type expectedErr struct {
+		wraps error
+		pass  bool
+	}
+	type msgArgs struct {
+		sender         sdk.AccAddress
+		multiplierName string
+		denomsToClaim  []string
```

Comment:
denomsToClaim isn't used in this test

Reply:
oops, fixed in https://github.com/Kava-Labs/kava/pull/955/commits/1c244265bdbd8db4499d223ccf0183162a1603e9